sox.features.js

```diff
@@ -1632,135 +1617,122 @@
       // Description: Hides the "Love This Site?" (weekly newsletter) module from the sidebar
 
-      $('#sidebar #newsletter-ad').remove();
+      const loveThisSite = document.querySelector('#sidebar #newsletter-ad')
+      if (loveThisSite) loveThisSite.remove();
     },
 
     chatEasyAccess: function() {
       // Description: Adds options to give a user read/write/no access in chat from their user popup dialog
 
-      $(document).on('sox-chat-user-popup', (e, node) => {
-        const $node = $(node).parent();
-        const id = $node.find('a')[0].href.split('/')[4];
-
-        //NOTE: $(node) !== $node
-        if ($(node).find('.chatEasyAccess').length) return;
-        if ($('.chatEasyAccess').length) $('.chatEasyAccess').remove();
-
-        //NOTE: $(node) !== $node
-        $(node).append('<div class="chatEasyAccess">give <b id="read-only">read</b> / <b id="read-write">write</b> / <b id="remove">no</b> access</div>');
-        $(document).on('click', '.chatEasyAccess b', function() {
-          const $that = $(this);
-          $.ajax({
-            url: 'https://chat.stackexchange.com/rooms/setuseraccess/' + location.href.split('/')[4],
-            type: 'post',
-            data: {
-              'fkey': window.fkey().fkey,
-              'userAccess': $that.attr('id'),
-              'aclUserId': id,
-            },
-            success: function(d) {
-              if (d === '') {
-                alert('Successfully changed user access');
-              } else {
-                alert(d);
-              }
-            },
+      const currentUserObject = sox.Chat.RoomUsers.current();
+      if (!(currentUserObject.is_owner || currentUserObject.is_moderator)) return; // only ROs and mods can grant access to other users
+      const accessButtons = '<div class="chatEasyAccess">give <b id="read-only">read</b> / <b id="read-write">write</b> / <b id="remove">no</b> access</div>';
+
+      window.addEventListener('sox-chat-user-popup', () => {
+        const node = document.querySelector('.popup.user-popup');
+        const nodeParent = node.parentElement;
+        const id = nodeParent.querySelector('a').href.split('/')[4];
+
+        if (nodeParent.querySelector('.chatEasyAccess')) return;
+        if (document.querySelector('.chatEasyAccess')) document.querySelector('.chatEasyAccess').remove();
+
+        node.insertAdjacentHTML('beforeend', accessButtons);
+        $(document).on('click', '.chatEasyAccess b', async function() {
+          const data = new FormData();
+          data.append('fkey', fkey().fkey);
+          data.append('userAccess', this.id);
+          data.append('aclUserId', id);
+          const response = await fetch('https://' + window.location.host + '/rooms/setuseraccess/' + location.href.split('/')[4], {
+            method: 'POST',
+            body: data
           });
+          const text = await response.text();
+          Notifier().notify(text === '' ? 'Successfully changed user access' : text);
```

Comment:
Is `Notifier` a StackExchange global?

Reply:
Yes, it exists in chat. You can try it with `Notifier().notify('hello');` for example.